### PR TITLE
refactor(add analysis filters and tests)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Adding filters for the analysis models and tests associated with the filers.

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/store/filters/test_status_analyses_filters.py [hunk 1]
```diff
@@ -275,9 +279,166 @@ def test_order_analyses_by_uploaded_at(
...
     assert old_analysis == analyses.all()[0]
     assert new_analysis == analyses.all()[1]
+
+
+def test_filter_analysis_by_case(base_store: Store, helpers: StoreHelpers, case_obj: Family):
+    """Test filtering of analyses by case."""
+
+    # GIVEN a set of mock analyses
+    analysis: Analysis = helpers.add_analysis(store=base_store)
+    analysis_other_case: Analysis = helpers.add_analysis(store=base_store, case=case_obj)
```
Comment (diitaz93) [inline tests/store/filters/test_status_analyses_filters.py:295]:
```suggestion
    analysis_other_case: Analysis = helpers.add_analysis(store=base_store, case=case_obj)
    assert analysis_other_case.family == case_obj
```

File: tests/store/filters/test_status_analyses_filters.py [hunk 2]
```diff
@@ -275,9 +279,166 @@ def test_order_analyses_by_uploaded_at(
...
+
+    # GIVEN an analysis query
+    analyses_query: Query = base_store._get_query(table=Analysis)
+
+    # WHEN filtering the analyses by case
+    analyses: Query = filter_analyses_by_case(analyses_query, case=case_obj)
```
Comment (diitaz93) [inline tests/store/filters/test_status_analyses_filters.py:301]:
```suggestion
    filtered_analyses: Query = filter_analyses_by_case(analyses_query, case=case_obj)
```

File: tests/store/filters/test_status_analyses_filters.py [hunk 3]
```diff
@@ -275,9 +279,166 @@ def test_order_analyses_by_uploaded_at(
...
+
+    # ASSERT that analyses is a query
+    assert isinstance(analyses, Query)
+
+    # THEN only the analysis belonging to the case should be retrieved
+    assert analysis not in analyses
+    assert analysis_other_case in analyses
+    assert analysis_other_case.family == case_obj
```
Comment (diitaz93) [inline tests/store/filters/test_status_analyses_filters.py:309]:
```suggestion
    assert analysis not in filtered_analyses
    assert analysis_other_case in filtered_analyses
```

File: tests/store/filters/test_status_analyses_filters.py [hunk 4]
```diff
@@ -275,9 +279,166 @@ def test_order_analyses_by_uploaded_at(
...
+    analysis: Analysis = helpers.add_analysis(store=base_store)
+    analysis_other_case: Analysis = helpers.add_analysis(store=base_store, case=case_obj)
+
+    # GIVEN an analysis query
+    analyses_query: Query = base_store._get_query(table=Analysis)
+
+    # WHEN filtering the analyses by case
+    analyses: Query = filter_analyses_by_case(analyses_query, case=case_obj)
+
+    # ASSERT that analyses is a query
```
Comment (diitaz93) [inline tests/store/filters/test_status_analyses_filters.py:303]:
```suggestion
    # THEN filtered_analyses is a query
```

File: cg/store/filters/status_analysis_filters.py [hunk 5]
```diff
@@ -28,7 +30,7 @@ def filter_not_completed_analyses(analyses: Query, **kwargs) -> Query:
     return analyses.filter(Analysis.completed_at.is_(None))
 
 
-def filter_uploaded_analyses(analyses: Query, **kwargs) -> Query:
+def filter_filter_uploaded_analyses(analyses: Query, **kwargs) -> Query:
```
[the remark this answers is not in the document]
Comment (ChrOertlin) [reply cg/store/filters/status_analysis_filters.py:33]: typo , fixed

File: tests/store/filters/test_status_analyses_filters.py [hunk 6]
```diff
@@ -42,15 +48,15 @@ def test_filter_valid_analyses_in_production(
     # WHEN retrieving valid in production analyses
     analyses: Query = filter_valid_analyses_in_production(analyses_query)
```
[the remark this answers is not in the document]
Comment (ChrOertlin) [reply tests/store/filters/test_status_analyses_filters.py:49]: In all other tests I am using just samples, pools etc. for the naming. I will keep analyses for now as well to keep consistency.

**Review by diitaz93 — APPROVED** [review]
👍

**Comment by ChrOertlin** [comment]
Stage:
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/hiseq.clinical
done.
Log deploy... done.
Prod:
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/hiseq.clinical
done.
Log deploy... done.
cg, version 27.1.31

**Pull request merged**